modules/http/src/main/scala/ch/epfl/bluebrain/nexus/commons/http/HttpClient.scala

```diff
@@ -57,6 +60,29 @@ object HttpClient {
     */
   type UntypedHttpClient[F[_]] = HttpClient[F, HttpResponse]
 
+  /**
+    * Interface syntax to expose new functionality into [[F[HttpResponse]]] type
+    *
+    * @param resp the [[HttpResponse]] wrapped in ''F[_]''
+    */
+  implicit class HttpResponseSyntax[F[_]](resp: F[HttpResponse])(implicit F: MonadError[F, Throwable],
+                                                                 cl: UntypedHttpClient[F]) {
+
+    /**
+      * Discards the response's bytes of the response's status matches some of ''expectedCodes''
+      * or triggers ''onFailure'' otherwise
+      *
+      * @param expectedCodes the codes to verify against the response code
+      * @param onFailure     the function to run when the response code does not match ''expectedCodes''
+      */
+    def discardOnCodesOr(expectedCodes: Set[StatusCode])(onFailure: => (HttpResponse) => F[Unit]): F[Unit] = {
```

Comment:
is there a reason for the call by name on the function? `onFailure: => (HttpResponse) => F[Unit]` ?

Reply:
It just seemed the right to me, given the use case. Usually when the method is `blablaOrBla` the signature it is like that. Like `option.getOrElse` `def getOrElse[B >: A](default: => B): B`

Reply:
the reason option.getOrElse has that signature is because it doesn't take a function as an argument, but just a value; to avoid computing the value when it's not required it's used as call by name; this is also to improve usability when the user has a value instead (you can use option.getOrElse(1) instead of option.getOrElse(_ => 1))

in your case you take a function as an argument, which will not be invoked unless required so the lazy evaluation is already handled by the fact that you explicitly require a function.